test/test_torch.py

```diff
@@ -6947,15 +6947,89 @@ def compare_strides(s1, s2, div):
 
     @skipMeta
     @dtypes(*torch.testing.get_all_dtypes())
-    def test_dlpack_conversion(self, device, dtype):
+    def test_dlpack_capsule_conversion(self, device, dtype):
         # DLpack does not explicitly support bool
         # It does it through uint8 type
         if dtype is torch.bool:
             return
+        if 'xla' in device:
+            return
         x = make_tensor((5,), device, dtype, low=-9, high=9)
         z = from_dlpack(to_dlpack(x))
         self.assertEqual(z, x)
 
+    @skipMeta
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_dlpack_protocol_conversion(self, device, dtype):
+        # DLpack does not explicitly support bool
+        # It does it through uint8 type
+        if dtype is torch.bool:
+            return
+        if 'xla' in device:
+            return
+        x = make_tensor((5,), device, dtype, low=-9, high=9)
+        z = from_dlpack(x)
+        self.assertEqual(z, x)
+
+    @skipMeta
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_dlpack_conversion_with_streams(self, device, dtype):
+        # DLpack does not explicitly support bool
+        # It does it through uint8 type
+        if dtype is torch.bool:
+            return
+        if 'xla' in device:
+            return
+        # Create a stream where the tensor will reside
+        if device == 'cuda':
+            x = make_tensor((5,), device, dtype, low=-9, high=9)
+            stream = torch.cuda.Stream()
+            with torch.cuda.stream(stream):
+                assert stream.query()
+                z = from_dlpack(x)
+                assert not stream.query()
+            assert stream.query()
+            self.assertEqual(z, x)
+
+    @skipMeta
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_dlpack_conversion_with_diff_streams(self, device, dtype):
+        # DLpack does not explicitly support bool
+        # It does it through uint8 type
+        if dtype is torch.bool:
+            return
+        if 'xla' in device:
+            return
+        if device == 'cuda':
+            from torch._C import _from_dlpack
+            x = make_tensor((5,), device, dtype, low=-9, high=9)
+            stream_a = torch.cuda.Stream()
+            stream_b = torch.cuda.Stream()
+            with torch.cuda.stream(stream_a):
+                assert stream_a.query()
+                assert stream_b.query()
+                z = _from_dlpack(x.__dlpack__(stream_b.cuda_stream))
```

Comment:
Same question here: is `x` guaranteed to be ready when we enter the scope of `stream_a`? I would have expected this snippet instead:
```python
with torch.cuda.stream(stream_a):
    x = make_tensor((5,), device, dtype, low=-9, high=9)
with torch.cuda.stream(stream_b):
    z = from_dlpack(x)
```
But maybe there's some assumptions in PyTorch's stream context that I missed?